lib/spack/spack/cmd/buildcache.py

```diff
@@ -255,6 +264,9 @@ def install_tarball(spec, args):
     force = False
     if args.force:
         force = True
+    unsigned = True
+    if args.unsigned:
```

Comment:
@gartung It looks like `unsigned` will always be true here based on how this is set. Something like

```
unsigned = False
if args.unsigned:
    unsigned = True
```

Which can be shortened:

```
unsigned = bool(args.unsigned)
```

(IMO the separate `unsigned` variable doesn't have to exist but that's fine either way)

I'm going to start looking through this in more detail but I figured I'd start with that

Reply:
Does this work is args.unsigned is undefined?

Reply:
`args.unsigned` will be defined since you added it as an option, it will be assigned a default (which I think is False, but you can ensure that by setting `default=False` in the call to `add_argument`)

Reply:
Shouldn't I default to True if -t is used?

Reply:
I don't follow: `-u` is available for `create`/`install`, `-t` is available for `keys`; so the `-t` option is not used here.

Also, these are flags so the question is: what does it mean when the user does `bincache install` vs. `bincache install --unsigned`? From looking at `build_tarball` it looks like when `unsigned == True` you want to avoid signing things, which makes sense to me. The name `unsigned` gives me the impression that when the user adds `--unsigned` that it should not be signed. So for those reasons I thought `unsigned` would default to false.

Sorry if that was overly-wordy: I'm having trouble explaining myself on this.

Reply:
I changed the initial value. Setting defaults is confusing me.

Reply:
I meant either -u --unsigned or -t --trust. I am reading up trying to figure out what default is set.

Reply:
It looks like `store_true` creates a default value of `False` (and `store_false` gives a default value of `True`): https://docs.python.org/2/library/argparse.html (15.4.3.2)

That being said IMO it's still clearer to assign the default explicitly

Reply:
Yes I was just looking at that page.